calico_kubernetes/policy.py

```diff
@@ -98,33 +98,36 @@ def _generate_rule(self, arguments):
         if protocol:
             rule_args["protocol"] = protocol
         if src_ports:
-            rule_args["src_ports"] = self._validate_ports(src_ports)
+            rule_args["src_ports"] = self._sanitize_ports(src_ports)
         if dst_ports:
-            rule_args["dst_ports"] = self._validate_ports(dst_ports)
+            rule_args["dst_ports"] = self._sanitize_ports(dst_ports)
         if icmp_type:
-            rule_args["icmp_type"] = self._validate_int(icmp_type)
+            rule_args["icmp_type"] = icmp_type
         if icmp_code:
-            rule_args["icmp_code"] = self._validate_int(icmp_code)
+            rule_args["icmp_code"] = icmp_code
         if src_net:
-            rule_args["src_net"] = self._validate_ip(src_net)
+            rule_args["src_net"] = src_net
         if src_label:
             rule_args["src_tag"] = self._validate_label(src_label)
         if dst_net:
-            rule_args["dst_net"] = self._validate_ip(dst_net)
+            rule_args["dst_net"] = dst_net
         if dst_label:
             rule_args["dst_tag"] = self._validate_label(dst_label)
 
         return Rule(**rule_args)
 
-    def _validate_int(self, value):
+    def _sanitize_ports(self, value):
         """
-        Validates the given value can be parsed as an int.
-        TODO: Move validation code into libcalico
+        Splits the given string of ports, delimited by a comma, into a list.
+        Removes all whitespace before splitting.
+        Validation is done in libcalico.
+
+        :param value: Port string.  E.g. "80,8001"
+        :return: List of ports.  E.g [80, 8001]
         """
-        try:
-            return int(value)
-        except ValueError:
-            raise PolicyParseError("Failed to parse %s, expecting int", value)
```

Comment:
I'm wondering about the PolicyParseError now.  `libcalico` throws ValueErrors, so I wonder if we should be doing the same to be consistent.

At a minimum, PolicyParseError should inherit from ValueError.